### ISSUE TITLE
Error handling in all() appears broken in v2

### ISSUE THREAD

**Issue by Turbo87** [machty/ember-concurrency#400]
```js
import { all, task } from 'ember-concurrency';

class extends Component {
  @task(function* () {
    try {
      // this should behave roughly similar to: yield this.task2.perform();
      yield all([this.task2.perform()]);
    } catch (error) {
      console.log("I've caught an error!!1", error);
    }
  })
  task1;

  @task(function* () {
    throw new Error('boom');
  })
  task2;
}
```

results in:

<img width="577" alt="Bildschirmfoto 2021-02-16 um 09 48 31" src="https://user-images.githubusercontent.com/141300/108039700-bb709b80-703c-11eb-8f24-857d253cd10c.png">

In other words, it looks like the `all()` helper does not handle error properly anymore. With v1 this code worked fine, but with v2 the uncaught error handler is invoked.

/cc @maxfierke

**Comment by maxfierke** [comment]
Oof. Yeah, that looks like a bug to me! Will take a look tonight.

**Comment by maxfierke** [comment]
Fixed in v2.0.1

**Comment by Turbo87** [comment]
appears to work great. thanks a lot! :)
